extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/scrubberBar.js

```diff
@@ -0,0 +1,326 @@
+/********************************************************************
+  SCRUBBER BAR
+*********************************************************************/
+var React = require('react'),
+    ReactDOM = require('react-dom'),
+    ResizeMixin = require('../mixins/resizeMixin'),
+    Thumbnail = require('./thumbnail'),
+    ThumbnailCarousel = require('./thumbnailCarousel'),
+    CONSTANTS = require('../constants/constants');
+
+var ScrubberBar = React.createClass({
+  mixins: [ResizeMixin],
+
+  getInitialState: function() {
+    this.lastScrubX = null;
+    this.isMobile = this.props.controller.state.isMobile;
+    this.touchInitiated = false;
+
+    return {
+      scrubberBarWidth: 0,
+      playheadWidth: 0,
+      scrubbingPlayheadX: 0,
+      hoveringX: 0,
+      currentPlayhead: 0,
+      transitionedDuringSeek: false
+    };
+  },
+
+  componentWillMount: function() {
+    if (this.props.seeking) {
+      this.setState({transitionedDuringSeek: true});
+    }
+  },
+
+  componentDidMount: function() {
+    this.handleResize();
+  },
+
+  componentWillReceiveProps: function(nextProps) {
+    if (this.state.transitionedDuringSeek && !nextProps.seeking) {
+      this.setState({transitionedDuringSeek: false});
+    }
+  },
+
+  componentWillUnmount: function() {
+    if (!this.isMobile){
+      ReactDOM.findDOMNode(this).parentNode.removeEventListener("mousemove", this.handlePlayheadMouseMove);
+      document.removeEventListener("mouseup", this.handlePlayheadMouseUp, true);
+    }
+    else{
+      ReactDOM.findDOMNode(this).parentNode.removeEventListener("touchmove", this.handlePlayheadMouseMove);
+      document.removeEventListener("touchend", this.handlePlayheadMouseUp, true);
+    }
+  },
+
+  getResponsiveUIMultiple: function(responsiveView){
+    var multiplier = this.props.skinConfig.responsive.breakpoints[responsiveView].multiplier;
+    return multiplier;
+  },
+
+  handleResize: function() {
+    this.setState({
+      scrubberBarWidth: ReactDOM.findDOMNode(this.refs.scrubberBar).clientWidth,
+      playheadWidth: ReactDOM.findDOMNode(this.refs.playhead).clientWidth
+    });
+  },
+
+  handlePlayheadMouseDown: function(evt) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Function has a complexity of 11 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)